cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1051,13 +1128,16 @@ rmm::device_buffer reader::impl::decompress_page_data(
   hostdevice_vector<gpu_inflate_input_s> inflate_in(0, num_comp_pages, stream);
   hostdevice_vector<gpu_inflate_status_s> inflate_out(0, num_comp_pages, stream);
 
+  device_span<gpu_inflate_input_s> inflate_in_view(inflate_in.device_ptr(), inflate_in.size());
+  device_span<gpu_inflate_status_s> inflate_out_view(inflate_out.device_ptr(), inflate_out.size());
+
   size_t decomp_offset = 0;
   int32_t argc         = 0;
   for (const auto& codec : codecs) {
-    if (codec.second > 0) {
+    if (codec.num_pages > 0) {
```

Comment:
This change to a struct makes the code much easier to follow and harder to get wrong. I like this.